docs/develop/testing-cloe.rst

```diff
@@ -0,0 +1,180 @@
+    Each of these files are cloe-launch profiles (i.e., Conanfiles) which should
+    be used to test all of the stack files in the same directory.
+
+    Note that this may involve re-compilation of one or more parts of Cloe.
+    For example, one profile may use a configuration that has the server in
+    cloe-engine disabled, another may force a specifc boost version, and yet
+    another may use a different compiler or toolchain.
+
+    These can actually be used by Conan directly::
+
+        # mkdir virtualenv && cd virtualenv
+        # conan install ../profile_default.py
+        # source cloe_launch_env.sh
+        # source activate.sh
+        # source activate_run.sh
+
+    Using cloe-launch automates that process and adds some flags for a one-stop
+    shop.
+
+``config_*.json``
+    Each of these files provides some base configuration that can be re-used by
+    other tests. They are generally complete stack files that can be checked or
+    run with ``cloe-engine`` inside a virtual run environment.
+
+``controller_*.json``
+    The controller configuration is used in both NOP as-well-as VTD simulator
+    binding stack files. To reduce duplication, the controller specific
+    configuration is placed in its own partial configuration file. This file
+    can be checked inside a cloe shell, but it might fail::
+
+        # cloe-engine check tests/controller_basic.json
+        tests/controller_basic.json: cannot find a vehicle with the name 'default': no entity with that name has been defined
+
+``test_*.json``
+    Each of these files is used for a specific test in a BATS file. In the
+    simplest case, it only includes another stack file.
+
+``test_*.bats``
+    Test executions are written in Bash and these tests are written with the
+    BATS framework.
+
+``setup_*.bash``
+    These files contain definitions that are used in all specific BATS scripts.
+    It should not be used directly, as it is loaded in each bats script.
+
+
+Testing stack files with Cloe is automated with the
+`Bash Automated Testing System <https://github.com/bats-core/bats-core>`__ (BATS).
+This requires the *bats* tool to be installed::
+
+    # sudo apt-get install bats
+
+Once installed, BATS can be used with individual files or with the entire
+directory::
+
+    # cloe-launch shell -P tests/profile_default.py -- -c "bats tests"
+    ✓ Expect check success    : test_minimator_smoketest.json                  : c7a427e7-eb2b-4ae7-85ec-a35b7540d4aa
+    ✓ Expect run success      : test_minimator_smoketest.json                  : 7c67ceb9-3d1d-47e4-9342-0b39099c59d6
+    ✓ Expect check/run success: test_minimator_smoketest.json [ts=5ms]         : 57254185-5480-4859-b2a5-6c3a211a22e0
+    ✓ Expect check/run success: test_minimator_smoketest.json [ts=60ms]        : a0d4982f-8c02-4759-bc88-cc30a1ccbbf0
+    ✓ Expect check/run success: test_minimator_multi_agent_smoketest.json      : 90e440ec-e8bc-40bb-8d2a-de224ee872bb
+
+    5 tests, 0 failures
+
```

Comment:
Maybe mention that `make smoketest` can be used alternatively?

Reply:
done